src/angular-app/languageforge/lexicon/editor/editor.component.ts

```diff
@@ -213,6 +213,20 @@ export class LexiconEditorController implements angular.IController {
   returnToList(): void {
     this.saveCurrentEntry();
     this.setCurrentEntry();
+    let index;
+    let hasNewEntry;
+    for (let iteration = 0; iteration < this.entries.length; iteration++) {
+      if (this.entries[iteration].constructor.name === 'LexEntry') {
+        index = iteration;
+        hasNewEntry = true;
+        break;
+      }
+    }
+    if (hasNewEntry) {
+      this.entries.splice(index, 1);
+      this.visibleEntries.splice(index, 1);
+      this.filteredEntries.splice(index, 1);
```

Comment:
I think this might not be correct. A new entry is not always in the filtered results. I'm not sure the difference between entries and visibleEntries, but perhaps the same thing could be an issue there.

Perhaps the [deleteEntry](https://github.com/sillsdev/web-languageforge/pull/309/files#diff-0ef01218d7165787be944e1c43388d8cL462) function could be used instead? Right now it combines prompting the user and deleting the entry, so it would need to be split into two functions.